hws/src/dr_rule.py

```diff
@@ -40,9 +40,9 @@ def tree_print_stes(stes, prefix, expected_miss_addr):
         return _str
 
 
```

Comment:
This will break some code logic. like when resize feature is used, in this case some of the actions STE's won't be in matcher.action_ste_0_id, they will be in an old action ste_0_id which is not visible here, also for a future feature of jump to STE array this also will break the logic

Reply:
> i do not know of such optimization where we jump from some rule STE to another Table STE,
> we need to pass by the table anchor, is there something i am missing?

Yes, there are some discussions about avoiding some of the anchors to reduce hop count. Talk to e.g. Khalid Manaa or Alex Vesker and they should be able to tell you more about if/when that will happen.

During/after a resize operation, is there any record of the old/new ICM addresses? Perhaps we could turn this parameter into a list of addresses, which would include the original address plus any new resized addresses?

Reply:
I've dropped the patch "HWS: constrain rule link list chasing" from this pull request since it doesn't work correctly for the resizing use-case. We can continue work separately.